symengine/parser.cpp

```diff
@@ -196,6 +208,31 @@ class ExpressionParser
                                  parse_string(iter + 1, operator_end[iter]));
                     iter = operator_end[iter] - 1;
 
+                } else if (s[iter] == '=') {
+                    result = Eq(result,
+                                parse_string(iter + 1, operator_end[iter]));
+                    iter = operator_end[iter] - 1;
+
+                } else if (s[iter] == '<' and s[iter + 1] == '=') {
```

Comment:
So, ideally to fix this we should move from a character based approach to a "token" based approach, where each token is one or more characters. Everything proceeds the same way but instead of iterating over characters we iterate over tokens. Tokens are generated in the `parse_expr` stage. You can think of `**` being converted to `^` a tokenization itself (where we restrict tokens to be only characters). 

Do you think you can implement this? As we add more and more operators, using only single characters as tokens will become a problem (and we can soon run out of symbols). If you don't want to tackle this now, go ahead with the special symbol hacky approach.